Sources/CardPayments/CardClient.swift

```diff
@@ -127,11 +129,28 @@ public class CardClient: NSObject {
                     }
                 }
                 
-                let cardResult = CardResult(orderID: orderId, deepLinkURL: url)
-                self.notifySuccess(for: cardResult)
+                guard let url = url else {
+                    self.notifyFailure(with: CardClientError.missingDeeplinkgURLError)
+                    return
+                }
+                
+                if url.absoluteString.contains("state=undefined") && url.absoluteString.contains("code=undefined") {
```

Comment:
If we convert our URL to a [URLComponents](https://developer.apple.com/documentation/foundation/urlcomponents) type we can use it's built in [queryItems array](https://developer.apple.com/documentation/foundation/urlcomponents/1779966-queryitems) versus parsing from the raw string directly. 

Something like ...

```
let components = URLComponents(url: url, resolvingAgainstBaseURL: false)
for queryItem in components.queryItems {
    if queryItem.name == "error" {
        // do stuff
    }
}
                
```

Reply:
we can use getQueryStringParameter function that we copied from another module and check for nil value there.

Reply:
thanks team, excellent advice, done: https://github.com/paypal/paypal-ios/pull/232/commits/ba6397f0ae564fe7b9f9c97d53c3b40b31740d4b